smartsim/exp/ray/raycluster.py

```diff
@@ -0,0 +1,493 @@
+        ray_args=None,
+        launcher="slurm",
+        interface="eth0",
+        alloc=None,
+        dash_port=8265,
+    ):
+        self.dashboard_port = dash_port
+        self.batch_settings = None
+        self.files = None
+
+        run_args = init_default({}, run_args, dict)
+        ray_args = init_default({}, ray_args, dict)
+
+        ray_exe_args = self._build_ray_exe_args(
+            ray_port, ray_password, interface, ray_args
+        )
+
+        run_settings = self._build_run_settings(launcher, alloc, run_args, ray_exe_args)
+        super().__init__(name, path, run_settings)
+
+    def _build_ray_exe_args(self, ray_port, ray_password, interface, ray_args):
+
+        # python script that launches ray head node
+        starter_script = find_ray_stater_script()
+        ray_starter_args = [
+            starter_script,
+            f"+port={ray_port}",
+            f"+ifname={interface}",
+            f"+ray-exe={find_ray_exe()}",
+            f"+head",
+        ]
+
+        if ray_password:
+            ray_starter_args += [f"+redis-password={ray_password}"]
+
+        if "dashboard-port" in ray_args:
+            self.dashboard_port = int(ray_args["dashboard-port"])
+        ray_starter_args += [f"+dashboard-port={self.dashboard_port}"]
+
+        used = ["block", "redis-password", "start", "head", "port", "dashboard-port"]
+        extra_ray_args = []
+        for key, value in ray_args.items():
+            if key not in used:
+                extra_ray_args += [f"+ray-args=--{key}={value}"]
+        ray_starter_args += extra_ray_args
+
+        return " ".join(ray_starter_args)
+
+    def _build_run_settings(self, launcher, alloc, run_args, ray_exe_args):
+
+        if launcher == "slurm":
+            run_settings = self._build_srun_settings(alloc, run_args, ray_exe_args)
+        elif launcher == "pbs":
+            run_settings = self._build_pbs_settings(run_args, ray_exe_args)
+        else:
+            raise SSUnsupportedError("Only slurm, and pbs launchers are supported.")
+
+        run_settings.set_tasks(1)
+        run_settings.set_tasks_per_node(1)
+        return run_settings
+
+    def _build_pbs_settings(self, run_args, ray_args):
+
+        aprun_settings = AprunSettings("python", exe_args=ray_args, run_args=run_args)
+        aprun_settings.set_tasks(1)
+
+        return aprun_settings
+
+    def _build_srun_settings(self, alloc, run_args, ray_args):
+
+        delete_elements(run_args, ["oversubscribe"])
+
+        run_args["unbuffered"] = None
+
+        srun_settings = SrunSettings(
+            "python",
+            exe_args=ray_args,
+            run_args=run_args,
+            alloc=alloc,
+        )
+        srun_settings.set_nodes(1)
+        return srun_settings
+
+
+class RayWorker(SmartSimEntity):
+    def __init__(
+        self,
+        name,
+        path,
+        ray_password,
+        ray_port,
+        run_args=None,
+        ray_args=None,
+        interface="eth0",
```

Comment:
Wondering if the default interface should be ``ipogif0`` which is the high speed network on XC.

``eth0`` would be cloud and local....

Reply:
Good idea.